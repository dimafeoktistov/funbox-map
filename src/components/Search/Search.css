.search {
    padding: 10px 5px;
}

.button {
    margin-left: 30px;
}

.form {
    display: flex;
    align-items: center;
}