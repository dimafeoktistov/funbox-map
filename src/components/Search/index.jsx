import React, { useState } from "react";
import PropTypes from "prop-types";

const Search = ({ addPlace, ...props }) => {
  const [place, setPlace] = useState("");
  const handlePlaceChange = event => setPlace(event.target.value);

  const handleSubmit = e => {
    e.preventDefault();
    addPlace(place);
    setPlace('')
  }
  // addPlaceToMap(place);
  return (
    <form onSubmit={handleSubmit} {...props}>
      <input
        type="text"
        placeholder="Введите место"
        onChange={handlePlaceChange}
        value={place}
        required
      />
      <button type="submit">Добавить к маршруту</button>
    </form>
  );
};

Search.propTypes = {
  addPlace: PropTypes.func.isRequired
};

export default Search;
