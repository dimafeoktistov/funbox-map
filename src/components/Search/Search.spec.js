import React from "react";
import Search from "./index";
import renderer from "react-test-renderer";

test("рендерится без краша", () => {
  const component = renderer.create(<Search addPlace={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
