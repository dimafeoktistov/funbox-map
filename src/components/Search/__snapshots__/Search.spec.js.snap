// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`рендерится без краша 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root-1 search"
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="place"
    >
      Место маршрута
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="place"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Введите место"
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-79 MuiButton-root-53 MuiButton-contained-64 MuiButton-containedPrimary-65 MuiButton-raised-67 MuiButton-raisedPrimary-68 button"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "marginLeft": "10px",
      }
    }
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-54"
    >
      Добавить
    </span>
    <span
      className="MuiTouchRipple-root-82"
    />
  </button>
</form>
`;
